Remove redundant casts

There are several redundant casts across the code base leading to compilation warnings that look like this:

````
/odk/collect/collect_app/src/main/java/org/odk/collect/android/tasks/DownloadFormListTask.java:128: warning: [cast] redundant cast to Element
                Element xformElement = (Element) xformsElement.getElement(i);
````

Find all such instances and remove the redundant casts. This should be submitted as a pull request that ONLY makes this change.

--- a/collect_app/src/main/java/org/odk/collect/android/activities/FormChooserList.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/FormChooserList.java
@@ -112,7 +112,7 @@ public class FormChooserList extends FormListActivity implements DiskSyncListene
     @Override
     protected void onListItemClick(ListView listView, View view, int position, long id) {
         // get uri to form
-        long idFormsTable = ((SimpleCursorAdapter) getListAdapter()).getItemId(position);
+        long idFormsTable = getListAdapter().getItemId(position);
         Uri formUri = ContentUris.withAppendedId(FormsColumns.CONTENT_URI, idFormsTable);
 
         Collect.getInstance().getActivityLogger().logAction(this, "onListItemClick",

--- a/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java
@@ -1162,7 +1162,7 @@ public class FormEntryActivity extends Activity implements AnimationListener,
                 }
 
                 // Create 'save' button
-                ((Button) endView.findViewById(R.id.save_exit_button))
+                 endView.findViewById(R.id.save_exit_button)
                         .setOnClickListener(new OnClickListener() {
                             @Override
                             public void onClick(View v) {

--- a/collect_app/src/main/java/org/odk/collect/android/preferences/GooglePreferencesFragment.java
+++ b/collect_app/src/main/java/org/odk/collect/android/preferences/GooglePreferencesFragment.java
@@ -48,7 +48,7 @@ public class GooglePreferencesFragment extends PreferenceFragment {
                                 .toString());
                         String value =
                                 (String) ((ListPreference) preference).getEntryValues()[index];
-                        ((ListPreference) preference).setSummary(value);
+                        preference.setSummary(value);
                         return true;
                     }
                 });

--- a/collect_app/src/main/java/org/odk/collect/android/tasks/DownloadFormListTask.java
+++ b/collect_app/src/main/java/org/odk/collect/android/tasks/DownloadFormListTask.java
@@ -125,7 +125,7 @@ public class DownloadFormListTask extends AsyncTask<Void, String, HashMap<String
                     // e.g., whitespace (text)
                     continue;
                 }
-                Element xformElement = (Element) xformsElement.getElement(i);
+                Element xformElement = xformsElement.getElement(i);
                 if (!isXformsListNamespacedElement(xformElement)) {
                     // someone else's extension?
                     continue;

--- a/collect_app/src/main/java/org/odk/collect/android/views/MediaLayout.java
+++ b/collect_app/src/main/java/org/odk/collect/android/views/MediaLayout.java
@@ -14,7 +14,6 @@
 
 package org.odk.collect.android.views;
 
-import android.app.Activity;
 import android.content.ActivityNotFoundException;
 import android.content.Context;
 import android.content.Intent;
@@ -25,8 +24,8 @@ import android.media.MediaPlayer;
 import android.media.MediaPlayer.OnCompletionListener;
 import android.net.Uri;
 import android.support.v7.widget.AppCompatImageButton;
-import android.util.Log;
 import android.util.DisplayMetrics;
+import android.util.Log;
 import android.view.View;
 import android.view.View.OnClickListener;
 import android.widget.CheckBox;
@@ -143,7 +142,7 @@ public class MediaLayout extends RelativeLayout implements OnClickListener {
             Intent i = new Intent("android.intent.action.VIEW");
             i.setDataAndType(Uri.fromFile(videoFile), "video/*");
             if (i.resolveActivity(getContext().getPackageManager()) != null) {
-                ((Activity) getContext()).startActivity(i);
+                getContext().startActivity(i);
             } else {
                 ToastUtils.showShortToast(getContext().getString(R.string.activity_not_found, "view video"));
             }

--- a/collect_app/src/main/java/org/odk/collect/android/widgets/AudioWidget.java
+++ b/collect_app/src/main/java/org/odk/collect/android/widgets/AudioWidget.java
@@ -167,7 +167,7 @@ public class AudioWidget extends QuestionWidget implements IBinaryWidget {
                         + mBinaryName);
                 i.setDataAndType(Uri.fromFile(f), "audio/*");
                 try {
-                    ((Activity) getContext()).startActivity(i);
+                    getContext().startActivity(i);
                 } catch (ActivityNotFoundException e) {
                     Toast.makeText(
                             getContext(),

--- a/collect_app/src/main/java/org/odk/collect/android/widgets/DateWidget.java
+++ b/collect_app/src/main/java/org/odk/collect/android/widgets/DateWidget.java
@@ -178,7 +178,7 @@ public class DateWidget extends QuestionWidget {
         if (mPrompt.getAnswerValue() != null) {
             DateTime ldt =
                     new DateTime(
-                            ((Date) ((DateData) mPrompt.getAnswerValue()).getValue()).getTime());
+                            ((Date) mPrompt.getAnswerValue().getValue()).getTime());
             mDatePicker.init(ldt.getYear(), ldt.getMonthOfYear() - 1, ldt.getDayOfMonth(),
                     mDateListener);
         } else {

--- a/collect_app/src/main/java/org/odk/collect/android/widgets/ExPrinterWidget.java
+++ b/collect_app/src/main/java/org/odk/collect/android/widgets/ExPrinterWidget.java
@@ -14,7 +14,6 @@
 
 package org.odk.collect.android.widgets;
 
-import android.app.Activity;
 import android.content.ActivityNotFoundException;
 import android.content.Context;
 import android.content.Intent;
@@ -175,7 +174,7 @@ public class ExPrinterWidget extends QuestionWidget implements IBinaryWidget {
         Collect.getInstance().getActivityLogger().logInstanceAction(this, "launchPrinter",
                 intentName, mPrompt.getIndex());
         Intent i = new Intent(intentName);
-        ((Activity) getContext()).startActivity(i);
+        getContext().startActivity(i);
 
         String[] splits;
         if (s != null) {
@@ -213,7 +212,7 @@ public class ExPrinterWidget extends QuestionWidget implements IBinaryWidget {
         //send the printDataBundle to the activity via broadcast intent
         Intent bcastIntent = new Intent(intentName + ".data");
         bcastIntent.putExtra("DATA", printDataBundle);
-        ((Activity) getContext()).sendBroadcast(bcastIntent);
+        getContext().sendBroadcast(bcastIntent);
     }
 
     @Override

--- a/collect_app/src/main/java/org/odk/collect/android/widgets/GeoShapeWidget.java
+++ b/collect_app/src/main/java/org/odk/collect/android/widgets/GeoShapeWidget.java
@@ -142,7 +142,7 @@ public class GeoShapeWidget extends QuestionWidget implements IBinaryWidget {
     @Override
     public void setBinaryData(Object answer) {
         // TODO Auto-generated method stub
-        String s = (String) answer.toString();
+        String s =  answer.toString();
         mStringAnswer.setText(s);
         mAnswerDisplay.setText(s);
         Collect.getInstance().getFormController().setIndexWaitingForData(null);

--- a/collect_app/src/main/java/org/odk/collect/android/widgets/SelectOneSearchWidget.java
+++ b/collect_app/src/main/java/org/odk/collect/android/widgets/SelectOneSearchWidget.java
@@ -142,7 +142,7 @@ public class SelectOneSearchWidget extends QuestionWidget implements
                 if (selectedTag == -1 && mSearchedItems.get(i).getValue().equals(s)) {
                     r.setChecked(true);
                     selectedTag = (Integer) r.getTag();
-                } else if (selectedTag.equals((Integer) r.getTag())) {
+                } else if (selectedTag.equals(r.getTag())) {
                     r.setChecked(true);
                 }
 

--- a/collect_app/src/main/java/org/odk/collect/android/widgets/TimeWidget.java
+++ b/collect_app/src/main/java/org/odk/collect/android/widgets/TimeWidget.java
@@ -59,7 +59,7 @@ public class TimeWidget extends QuestionWidget {
             // create a new date time from date object using default time zone
             DateTime ldt =
                     new DateTime(
-                            ((Date) ((TimeData) prompt.getAnswerValue()).getValue()).getTime());
+                            ((Date) prompt.getAnswerValue().getValue()).getTime());
             System.out.println("retrieving:" + ldt);
             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                 mTimePicker.setHour(ldt.getHourOfDay());

--- a/collect_app/src/main/java/org/odk/collect/android/widgets/VideoWidget.java
+++ b/collect_app/src/main/java/org/odk/collect/android/widgets/VideoWidget.java
@@ -206,7 +206,7 @@ public class VideoWidget extends QuestionWidget implements IBinaryWidget {
                         + mBinaryName);
                 i.setDataAndType(Uri.fromFile(f), "video/*");
                 try {
-                    ((Activity) getContext()).startActivity(i);
+                    getContext().startActivity(i);
                 } catch (ActivityNotFoundException e) {
                     Toast.makeText(
                             getContext(),

